src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3167,7 +3167,7 @@ protected override IPointerTypeSymbol CommonCreatePointerTypeSymbol(ITypeSymbol
             return NamedTypeSymbol.CreateTuple(
                 locationOpt: null, // no location for the type declaration
                 elementTypesWithAnnotations: typesBuilder.ToImmutableAndFree(),
-                elementLocations: elementLocations,
+                elementLocations: elementLocations!,
```

Comment:
>elementLocations [](start = 34, length = 16)

Should the type of `elementLocations` be adjusted in public API then, so we avoid the suppression?

Reply:
I don't think so. If you're creating this from a public context, you should pass real locations as more of a matter of principle.

---
In reply to: [367179639](https://github.com/dotnet/roslyn/pull/40946#discussion_r367179639) [](ancestors = 367179639)

Reply:
Just to clarify: When do we need to pass `null` locations internal?
Maybe we can mark as non-nullable across the board...

---
In reply to: [370338279](https://github.com/dotnet/roslyn/pull/40946#discussion_r370338279) [](ancestors = 370338279,367179639)